server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/DomainAliasService.java

```diff
@@ -81,6 +82,7 @@ public boolean hasAliases(Domain domain) throws DomainListException, RecipientRe
     }
 
     public void addDomainAlias(Domain sourceDomain, Domain destinationDomain) throws DomainListException, RecipientRewriteTableException {
+        checkCyclicMapping(sourceDomain, destinationDomain);
```

Comment:
IMO this should be a responsibility of the backend component and not of the backend layer.